kotest-framework/kotest-framework-engine/src/commonMain/kotlin/io/kotest/engine/spec/SpecExtensions.kt

```diff
@@ -40,18 +44,21 @@ internal class SpecExtensions(private val registry: ExtensionRegistry) {
          registry.all() // globals
    }
 
+   private inline fun <reified T> extensionsOf(spec: Spec) = extensions(spec).filterIsInstance<T>()
+
    /**
-    * Runs all the [BeforeSpecListener] for this [Spec]. All errors are caught and wrapped
+    * Runs all the [BeforeSpecListener]s for this [Spec]. All errors are caught and wrapped
     * in [ExtensionException.BeforeSpecException] and if more than one error,
     * all will be wrapped in a [MultipleExceptions].
+    *
+    * [BeforeSpecListener]s are invoked in a child coroutine of the given [specCoroutineContext].
     */
-   suspend fun beforeSpec(spec: Spec): Result<Spec> {
+   fun beforeSpec(spec: Spec, specCoroutineContext: CoroutineContext): Result<Spec> {
       logger.log { Pair(spec::class.bestName(), "beforeSpec $spec") }
-
-      val errors = extensions(spec)
-         .filterIsInstance<BeforeSpecListener>()
+      val scope = CoroutineScope(specCoroutineContext + CoroutineName("before-spec-scope"))
+      val errors = extensionsOf<BeforeSpecListener>(spec)
          .mapNotNull { ext ->
-            runCatching { ext.beforeSpec(spec) }
+            runCatching { scope.launch { ext.beforeSpec(spec) } }
```

Comment:
```suggestion
            runCatching { withContext(scope) { ext.beforeSpec(spec) } }
```

`runCatching` waits. Launching an extra job would obscure the fact that we're sequentially executing a suspend function.

Reply:
Correction: `runCatching` is not a coroutine scoping function. In this case it would just check that the launch succeeds (without anything running at this time). This is not what's intended, right?

Reply:
Correct, no.

Reply:
withContext(scope)  doesn't seem to compile because it expects a context not a scope.

Reply:
Right, the whole concept cannot work. There would be two "parents" expecting the same coroutine (`ext.beforeSpec(spec)`) to complete:
1. The parent coroutine `specCoroutineContext`.
2. The blocking scope (`runCatching`, `mapNotNull`, ...).

Reply:
Makes sense. I am going to re-work this now I merged the spec runners together and simplified that part of the stack a lot.